### PR TITLE
prevent bad pod.status.phase transitions

### PR DESCRIPTION
**Pull request by deads2k**
A thought for how to catch and stop bad transitions in a spot where we have user information and can easily emit an event.

@openshift/sig-master @sjenning

### CODE REVIEW

File: pkg/pod/admission/rejectbadstatus/reject_bad_pod_status.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		q.eventRecorder.Event(oldPod, v1.EventTypeWarning, "InvalidPodPhaseTransition", err.Error())
+		klog.Errorf("invalid pods/status transition for %s[%s] from %v: %v", a.GetName(), a.GetNamespace(), spew.Sdump(a.GetUserInfo()), err)
+		return admission.NewForbidden(a, err)
+	}
+
+	return nil
+}
+
+// allowedPhaseTransitions maps from current phase to the list of allowed phases
+var allowedPhaseTransitions = map[string]sets.String{
```
Comment (deads2k) [inline pkg/pod/admission/rejectbadstatus/reject_bad_pod_status.go]: @sjenning I don't know the valid phase transitions.

**Comment by deads2k** [comment]
terraform
/retest

**Comment by deads2k** [comment]
terraform

/retest

**Comment by tnozicka** [comment]
If I recall correctly @sjenning has added an admission in kube last time, might just need tweaking or making sure it is setup correctly

/hold
(until we sync on where to check it)

**Pull request closed without merging**